### PR TITLE
rtl_433: add package

### PR DESCRIPTION
**Pull request by TimelessNL**
Maintainer: Jasper Scholte/ @TimelessNL
Compile tested: (x86 / brcm2708 /  ar71xx, GEODE / Raspberry3B / GL.iNet 6416, LEDE 17.01.2)
Run tested: (x86 / brcm2708 /  ar71xx, GEODE / Raspberry3B / GL.iNet 6416, LEDE 17.01. Tested (running) for more than 3 months)

Description:
This will add the rtl_433 package. I've been running this since a couple of months. And so far no issues. It's nice to have in combination with Domoticz.

### CODE REVIEW

**Comment by pprindeville** [comment]
Please change commit message header to:

`rtl_433: add package`

File: utils/rtl_433/Makefile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  MAINTAINER:=Jasper Scholte <NightNL@outlook.com>
+endef
+
+define Package/rtl-sdr/description
+  rtl_433 turns your Realtek RTL2832 based DVB dongle into a 433.92MHz generic data receiver.
+endef
+
+TARGET_CFLAGS += $(FPIC)
+
+define Build/InstallDev
```
Comment (pprindeville) [inline utils/rtl_433/Makefile]: Why is this needed?

Comment (TimelessNL) [reply utils/rtl_433/Makefile]: Indeed. It's not used as a library. I just copied some stuff from rtl-sdr. Sorry

**Comment by TimelessNL** [comment]
I've reworked your comments.

File: utils/rtl_433/Makefile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=rtl_433
+PKG_VERSION:=6531ea4
+PKG_RELEASE:=$(PKG_SOURCE_VERSION)
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=git://github.com/merbanan/rtl_433.git
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE_VERSION:=6531ea48a7933ac8289724672059e54fd8aad8eb
```
Comment (pprindeville) [inline utils/rtl_433/Makefile]: Need:

```
WARNING: PKG_MIRROR_HASH is missing, set to f96301c8d7e87605b113aad4493f15b63a675fea6f00642b0606de5a3a858746
```

Comment (TimelessNL) [reply utils/rtl_433/Makefile]: Ok will do

File: utils/rtl_433/Makefile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+# Copyright (C) 2017 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=rtl_433
+PKG_VERSION:=6531ea4
+PKG_RELEASE:=$(PKG_SOURCE_VERSION)
```
Comment (pprindeville) [inline utils/rtl_433/Makefile]: Any reason to not go with numbered releases?

Comment (pprindeville) [reply utils/rtl_433/Makefile]: Last one was 15 months ago, but there's been plenty of activity in the meantime... Can you ping the maintainer about another release?

Comment (TimelessNL) [reply utils/rtl_433/Makefile]: Yeah that's exactly the reason why. I also mentioned it in my commit message.
I could ping the maintainer see if he has plans for a release. But if not does that have any impact on my pull-request? I hope not, it's a very nice tool.

To be more precise I chose this version because I just opened a pull-request in Domoticz to support new sensors from rtl_433 and therefor I needed a up-to-date version of rtl_433.

File: utils/rtl_433/Makefile (hunk 2 above)
Comment (TimelessNL) [reply utils/rtl_433/Makefile]: done

File: utils/rtl_433/Makefile (hunk 3 above)
Comment (TimelessNL) [reply utils/rtl_433/Makefile]: I noticed this issue topic: https://github.com/merbanan/rtl_433/issues/470
The maintainer left a response quite some time ago, so he is probably aware of the release/versioning issue but has not come up with a solution yet :(

**Comment by TimelessNL** [comment]
@pprindeville, what do you think about the current state of this pull-request?

File: utils/rtl_433/Makefile (hunk 2 above)
Comment (pprindeville) [reply utils/rtl_433/Makefile]: I think `PKG_HASH` is sufficient.

Comment (TimelessNL) [reply utils/rtl_433/Makefile]: Is updated :)

Comment (pprindeville) [reply utils/rtl_433/Makefile]: Sorry, my mistake.  For retrieving a tarball, use `PKG_HASH`.  For retrieving a repository via `git` or `svn`, you do need `PKG_SOURCE_HASH`.  Don't ask me why.

So you can drop the previous commit in a rebase and force push.

Comment (TimelessNL) [reply utils/rtl_433/Makefile]: You didn't mean PKG_MIRROR_HASH?

Comment (pprindeville) [reply utils/rtl_433/Makefile]: Right, sorry.

**Comment by TimelessNL** [comment]
Thanks :)
